libtextsecure/account_manager.js

```diff
@@ -544,6 +546,40 @@
 
       this.dispatchEvent(new Event('registration'));
     },
+    async authoriseSecondaryDevice(secondaryDevicePubKey) {
+      if (secondaryDevicePubKey === textsecure.storage.user.getNumber()) {
+        throw new Error(
+          'Cannot register primary device pubkey as secondary device'
+        );
+      }
+
+      // Validate pubKey
+      const c = new Whisper.Conversation({
+        id: secondaryDevicePubKey,
+        type: 'private',
+      });
+      const validationError = c.validateNumber();
+      if (validationError) {
+        throw new Error('Invalid secondary device pubkey provided');
+      }
+      // Ensure there is a conversation existing
+      try {
+        await ConversationController.getOrCreateAndWait(
```

Comment:
Could we do `const c = ConversationController.getOrCreateAndWait`... and then validate from that rather than creating a whisper conversation directly?

Reply:
Yes, good point! Updated.